### PR TITLE
New protocol Tilt Analysis

### PR DESCRIPTION
**Pull request by danielmarchan3**
Protocol used to detect tilted micrographs by analyzing the PSD correlations of different segments of the image.

### CODE REVIEW

**Review by cossorzano — APPROVED** [review]

File: xmipp3/protocols/protocol_tilt_analysis.py [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+        form.addSection(label=Message.LABEL_INPUT)
+
+        form.addParam('inputMicrographs', PointerParam,
+                      pointerClass='SetOfMicrographs',
+                      label="Input micrographs", important=True,
+                      help='Select the SetOfMicrograph to be preprocessed.')
+
+        form.addParam('window_size', IntParam, label='Window size',
+                      default=1024, expertLevel=LEVEL_ADVANCED, validators=[GE(256, 'Error must be greater than 256')],
+                      help='''By default, the micrograph will be divided into windows of dimensions 512x512,
```
Comment (albertmena) [inline xmipp3/protocols/protocol_tilt_analysis.py:74]: Remove the break line after 512x512,

File: xmipp3/protocols/protocol_tilt_analysis.py [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+
+        form.addParam('window_size', IntParam, label='Window size',
+                      default=1024, expertLevel=LEVEL_ADVANCED, validators=[GE(256, 'Error must be greater than 256')],
+                      help='''By default, the micrograph will be divided into windows of dimensions 512x512, 
+                            the PSD and its correlations will be computed in every segment.''')
+
+        form.addParam('objective_resolution', FloatParam, label='Objective resolution',
+                      default=3, expertLevel=LEVEL_ADVANCED, validators=[GT(0, 'Error must be Positive')],
+                      help='''By default, micrographs PSD will be cropped into a central windows of dimensions
+                            (xdim*(sampling rate/objective resolution))
```
Comment (albertmena) [inline xmipp3/protocols/protocol_tilt_analysis.py:80]: Remove the break line

File: xmipp3/protocols/protocol_tilt_analysis.py [hunk 3]
```diff
@@ -0,0 +1,670 @@
...
+        elif not self.finished:
+            # If we are not finished and no new output have been produced
+            # it does not make sense to proceed and updated the outputs
+            # so we exit from the function here
+            return
+
+        micsAccepted = []
+        micsDiscarded = []
+        for mic in newDone:
+            id = mic.getObjId()
```
Comment (albertmena) [inline xmipp3/protocols/protocol_tilt_analysis.py:220]: Rename this variable: https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=544&id=ScipionEmXmipp&open=AYHc0xvfDhQWe92u0a3B

File: xmipp3/protocols/protocol_tilt_analysis.py [hunk 4]
```diff
@@ -0,0 +1,670 @@
...
+        output_image = ImageHandler().createImage()
+        output_array = np.zeros((subWindStep*2, subWindStep*2))
+        ih = ImageHandler()
+
+        for i in range(0, 3, 2):
+            for j in range(0, 3, 2):
+                window = micImage.window2D(x_steps[i], y_steps[j], x_steps[i+1], y_steps[j+1])
+                x_dim, y_dim, z, n = window.getDimensions()
+                # NORMALIZED
+                mean, dev, min, max = window.computeStats()
```
Comment (albertmena) [inline xmipp3/protocols/protocol_tilt_analysis.py:383]: Rename all min, max variables: it shadows a builtin

File: xmipp3/protocols/protocol_tilt_analysis.py [hunk 5]
```diff
@@ -0,0 +1,670 @@
...
+        return os.path.join(micFolder, filename)
+
+
+    # --------------------------- INFO functions -------------------------------
+    def _summary(self):
+        fnSummary = self._getPath("summary.txt")
+        if not os.path.exists(fnSummary):
+            summary = ["No summary information yet."]
+        else:
+            fhSummary = open(fnSummary, "r")
```
Comment (albertmena) [inline xmipp3/protocols/protocol_tilt_analysis.py:587]: For many mics the summary will become unreadable. Simplify this or summary some general values as mean of the means ...

**Review by albertmena — APPROVED** [review]

**Pull request merged**